compiler/cpp/src/thrift/generate/t_py_generator.cc

```diff
@@ -2742,8 +2765,71 @@ string t_py_generator::type_name(t_type* ttype) {
   return ttype->get_name();
 }
 
+string t_py_generator::arg_hint(t_type* type) {
+  if(gen_type_hints_) {
+      return ": " + type_to_py_hint(type);
+  }
+
+  return "";
+}
+
+string t_py_generator::func_hint(t_type* type) {
+  if(gen_type_hints_) {
+      return " -> " + type_to_py_hint(type);
+  }
+
+  return "";
+}
+
+/**
+ * Converts the parse type to a Python type hint
+ */
+string t_py_generator::type_to_py_hint(t_type* type) {
+  return "typing.Optional[" + type_to_py_type(type) + "]";
+}
+
+/**
+ * Converts the parse type to a Python type
+ */
+string t_py_generator::type_to_py_type(t_type* type) {
+  type = get_true_type(type);
+
+  if (type->is_binary()) {
+    return  "bytes";
+  }
+  if (type->is_base_type()) {
+    t_base_type::t_base tbase = ((t_base_type*)type)->get_base();
+    switch (tbase) {
+    case t_base_type::TYPE_VOID:
+      return "None";
+    case t_base_type::TYPE_STRING:
+      return "str";
+    case t_base_type::TYPE_BOOL:
+      return "bool";
+    case t_base_type::TYPE_I8:
+    case t_base_type::TYPE_I16:
+    case t_base_type::TYPE_I32:
+    case t_base_type::TYPE_I64:
+      return "int";
+    case t_base_type::TYPE_DOUBLE:
+      return "float";
+    }
+  } else if (type->is_enum() || type->is_struct() || type->is_xception()) {
+    return type_name(type);
+  } else if (type->is_map()) {
+    return "dict[" + type_to_py_type(((t_map*)type)->get_key_type()) + ", " + type_to_py_type(((t_map*)type)->get_val_type()) + "]";
```

Comment:
This should be using `Dict[...]` and `from typing import Dict` ?

Same below for `Set` and `List` ?

Reply:
With Python 3.7 or newer, you can use dict as a generic type if you start your module with [from __future__ import annotations](https://www.python.org/dev/peps/pep-0563/), and as of Python 3.9, dict (as well as other standard containers) [supports being used as generic type even without that directive](https://www.python.org/dev/peps/pep-0585/).